subprojects/distributions-core/build.gradle.kts

```diff
@@ -15,6 +15,9 @@ dependencies {
     pluginsRuntimeOnly(project(":dependency-management")) {
         because("This is a core extension module (see DynamicModulesClassPathProvider.GRADLE_EXTENSION_MODULES)")
     }
+    pluginsRuntimeOnly(project(":java-platform")) {
+        because("This is a dependency-management module, that isn't java specific.")
```

Comment:
❓ I don't think this comment is right. Looking at the actual implementation of the `JavaPlatformPlugin`, it does JVM specific stuff, particularly setting the `Usage.JAVA_API` and `Usage.JAVA_RUNTIME` attributes on the outgoing variants. That's on top of the fact that there is `java` in the name of the Gradle project and the plugin name. 

For these reasons I would say this plugin does not belong in the core distribution. In general, IMO anything in the `jvm` platform does not belong in the core distribution. 

I think we are missing a `platform-base` sort of plugin which can create platforms generic to any ecosystem, but for now I think the `JavaPlatformPlugin` is too specialized for the core distribution.